.github/workflows/build.yml

```diff
@@ -0,0 +1,73 @@
+name: Build
+on: [push]
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}-latest
+    continue-on-error: ${{ matrix.continue-on-error }}
+    strategy:
+      matrix:
+        continue-on-error: [false]
+        # We test against Java releases:
+        #  - greater than 7
+        #  AND
+        #    - with LTS and in "Premier Support" phase
+        #    OR
+        #    - not succeeded by any other release
+        # No Oracle JDK >11, see
+        # https://blogs.oracle.com/java-platform-group/oracle-jdk-releases-for-java-11-and-later
+        # https://www.oracle.com/technetwork/java/java-se-support-roadmap.html
+        java:
+        # LTS and in "Premier Support" as of 2019-03 (until 2022-03)
+        - 8
+        # LTS and in "Premier Support" as of 2019-03 (until 2023-09)
+        - 11
+        # Not succeeded as of 2021-04. Scheduled to be succeeded by
+        # Java 17 on 2021-09-14, see http://openjdk.java.net/projects/jdk/17/
+        - 16
+        os: [ubuntu, windows]
+        include:
+        - continue-on-error: true
+          os: ubuntu
+          java: 17
+        - continue-on-error: true
+          os: macos
+          java: 16
+        - continue-on-error: true
+          os: windows
+          java: 17
+    env:
+      # We only post coverage data for exactly one build to coveralls.
+      # To pick that build from the build matrix, we designate one
+      # Java version to be the "main" one.
+      MAIN_JAVA: 11
+      # We only post coverage data for exactly one build to coveralls.
+      # To pick that build from the build matrix, we designate one
+      # OS to be the "main" one.
+      MAIN_OS: ubuntu
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - uses: actions/setup-java@v2
+      with:
+        java-version: ${{ matrix.java }}
+        distribution: 'adopt'
+        architecture: x64
+    - uses: actions/cache@preview
```

Comment:
If I understand actions correctly, then the `@preview` version of the cache action is from 2019 and very outdated by now. Why not use the current `v2` version?